pkg/services/alerting/service.go

```diff
@@ -116,6 +119,24 @@ func (s *AlertNotificationService) GetAlertNotificationsWithUid(ctx context.Cont
 }
 
 func (s *AlertNotificationService) UpdateAlertNotificationWithUid(ctx context.Context, cmd *models.UpdateAlertNotificationWithUidCommand) error {
+	var err error
+	cmd.EncryptedSecureSettings, err = s.EncryptionService.EncryptJsonData(ctx, cmd.SecureSettings, setting.SecretKey)
+	if err != nil {
+		return err
+	}
+
+	model := models.AlertNotification{
+		Uid:      cmd.NewUid,
```

Comment:
Think this should use the current UID and not the new one. New one not populated by provisioning, but might be via the HTTP API.

```suggestion
		Uid:      cmd.Uid,
```